src/arch/win32/win32socketinterf.h

```diff
@@ -35,6 +36,16 @@ class CWin32SocketInterface {
   private:
     CWin32SocketInterface(); //this function is not implemented as we don't want instances of this class
 
+    static LPWSTR fromCharString(const char *paValue) {
+      const size_t size = (strlen(paValue) + 1) * sizeof(wchar_t);
+      wchar_t *buf = new wchar_t[size];
+      buf[size] = L'\0';
+      MultiByteToWideChar(CP_UTF8, 0, paValue, -1, buf, (int)size);
+      const wchar_t *ret = (const wchar_t*)buf;
+      delete[] buf;
+      return (LPWSTR)ret;
```

Comment:
I think you are returning an invalid pointer here. `ret` points to the same memory as `buf`, but `buf`'s memory is deleted in line 45

Reply:
In my understanding, I have stored the buf contents to ret constant before deleting it so the returned value still has a valid conversion result.

Reply:
Unfortunately this is not the behavior. you stored in  line 44 in ret the pointer to the memory that you created. However in line 45 you delete that memory. therefore the pointer that you are returning is pointing to freed memory.

Reply:
@MartinMelikMerkumians @azoitl Now I create a new std::wstring as a result buffer.

Reply:
I added the following function for reverse conversion locally:

```
    static std::string fromWCharString(const wchar_t *paValue) {
      const int size = WideCharToMultiByte(CP_UTF8, 0, paValue, -1, nullptr, 0, nullptr, nullptr);
      char *buf = new char[size + 1];
      buf[size + 1] = '\0';
      WideCharToMultiByte(CP_UTF8, 0, paValue, -1, buf, size, nullptr, nullptr);
      std::string ret(buf);
      delete[] buf;
      return ret;
    }
```

No problem so far for:
```
DEVLOG_INFO("CWin32SocketInterface: Opening TCP-Server connection at: %s:%d\n", fromWCharString(fromCharString(paIPAddr).c_str()).c_str(), paPort);
```

Reply:
Finally, a conversion without buffer: https://learn.microsoft.com/id-id/archive/blogs/larryosterman/converting-win32-api-results-to-stdwstring-or-stdstring